testserver/testserver.go

```diff
@@ -403,8 +433,8 @@ func NewTestServer(opts ...TestServerOpt) (TestServer, error) {
 
 	if serverArgs.cockroachBinary != "" {
 		// CockroachBinaryPathOpt() overrides the flag or env variable.
-	} else if len(*customBinaryFlag) > 0 {
-		serverArgs.cockroachBinary = *customBinaryFlag
+	} else if len(GetCustomBinaryFlag()) > 0 {
+		serverArgs.cockroachBinary = GetCustomBinaryFlag()
```

Comment:
Please assign return-value of `GetCustomBinaryFlag()` to a local-variable and then re-use it across branch-check and serverArg assignment.

We have exclusion inside `GetCustomBinaryFlag()` not across the two invocations.